### PR TITLE
Auto specify filename and file meta for cli download

### PR DESCRIPTION
**Pull request by duykhang53**
- Auto specify filename and add metadata;
- Implement youtube video url for --id;
- Update README;
- Add .jshintrc;

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,18 +21,26 @@
...
+    "ffmetadata": "^1.6.0",
     "fluent-ffmpeg": "^2.1.2",
     "minimist": "^1.2.0",
     "mkdirp": "^0.5.1",
+    "qs": "^6.9.1",
+    "retriable-download": "^2.1.0",
+    "sanitize-filename": "^1.6.3",
     "through2": "^2.0.3",
+    "updated-youtube-info": "^1.4.7",
+    "youtube-info": "^1.3.2",
```
Comment (codealchemist) [inline package.json]: This one is currently deprecated and I it's not being used.
I think we need to remove it.

File: package.json [hunk 2]
```diff
@@ -21,18 +21,26 @@
   "bugs": {
     "url": "https://github.com/codealchemist/youtube-audio-server/issues"
   },
+  "files": ["!/temp-audio/**", "src/*", "!/src/test"],
   "homepage": "https://github.com/codealchemist/youtube-audio-server#readme",
   "dependencies": {
     "express": "^4.17.1",
     "express-no-favicons": "0.0.1",
+    "ffmetadata": "^1.6.0",
```
Comment (codealchemist) [inline package.json:33]: Could be better if we use `fluent-ffmpeg` to write metadata, which is already available.

File: package.json [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "youtube-audio-server",
-  "version": "2.4.1",
+  "version": "2.6.0",
   "description": "Easily stream and download audio from YouTube.",
   "main": "src/index.js",
   "scripts": {
-    "start": "./src/bin.js",
+    "start": "cross-env ./src/bin.js",
```
Comment (codealchemist) [inline package.json:7]: 🙌

File: package.json [hunk 4]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "youtube-audio-server",
-  "version": "2.4.1",
+  "version": "2.6.0",
```
Comment (codealchemist) [inline package.json]: The prev. version was `2.4.x` and we're adding new features on this one.
We should probably just bump the minor in 1.
```suggestion
  "version": "2.5.0",
```

File: package.json (hunk 1 above)
Comment (codealchemist) [reply package.json]: Oh, and I just find out we should be able to use `ytdl-core` to get video info:
https://www.npmjs.com/package/ytdl-core#async-ytdlgetinfourl-options

File: package.json [hunk 5]
```diff
@@ -21,18 +21,26 @@
...
+  "files": ["!/temp-audio/**", "src/*", "!/src/test"],
   "homepage": "https://github.com/codealchemist/youtube-audio-server#readme",
   "dependencies": {
     "express": "^4.17.1",
     "express-no-favicons": "0.0.1",
+    "ffmetadata": "^1.6.0",
     "fluent-ffmpeg": "^2.1.2",
     "minimist": "^1.2.0",
     "mkdirp": "^0.5.1",
+    "qs": "^6.9.1",
```
Comment (codealchemist) [inline package.json]: Newer versions of `ytdl-core` already support passing a video ID or URL so we don't need to parse the URL now, yay!

File: package.json [hunk 6]
```diff
@@ -21,18 +21,26 @@
...
+    "retriable-download": "^2.1.0",
+    "sanitize-filename": "^1.6.3",
     "through2": "^2.0.3",
+    "updated-youtube-info": "^1.4.7",
+    "youtube-info": "^1.3.2",
     "youtube-node": "^1.3.3",
-    "ytdl-core": "^0.29.2"
+    "ytdl-core": "^3.2.2"
```
Comment (codealchemist) [inline package.json]: In `master` we already have v4 😉

File: package.json [hunk 7]
```diff
@@ -21,18 +21,26 @@
...
   "homepage": "https://github.com/codealchemist/youtube-audio-server#readme",
   "dependencies": {
     "express": "^4.17.1",
     "express-no-favicons": "0.0.1",
+    "ffmetadata": "^1.6.0",
     "fluent-ffmpeg": "^2.1.2",
     "minimist": "^1.2.0",
     "mkdirp": "^0.5.1",
+    "qs": "^6.9.1",
+    "retriable-download": "^2.1.0",
```
Comment (codealchemist) [inline package.json:37]: Can we look into using something more widely used?
Like:
https://www.npmjs.com/package/download

**Comment by codealchemist** [comment]
Hi @duykhang53 !
First of all, thanks a lot for the PR!
I know this was ages ago and I'm sorry for not being able to pay attention to it sooner 🙏 
I love the idea of being able to read and store video info inside ID3 tags.
🙌

**Comment by duykhang53** [comment]
i'll leave package-lock.json to you

File: package.json (hunk 7 above)
Comment (duykhang53) [reply package.json:37]: still need retriable-download for cover art

Comment (codealchemist) [reply package.json:37]: We can do something like this:
`download(imgUrl, { retry: 3 })`

https://www.npmjs.com/package/download
It has 3.5M weekly downloads and a clear API.
Weights only 6KB.

https://www.npmjs.com/package/retriable-download
It has 48 weekly downloads and the same weight.
We should be safer with the first one for the same cost in KB 😄

Comment (codealchemist) [reply package.json:37]: BTW, that returns a stream we can then pipe to a `createWriteStream`.

Comment (duykhang53) [reply package.json:37]: nah, i just want to keep the source short, it's a simple feature btw

Comment (duykhang53) [reply package.json:37]: the purpose is just save the image to temp folder and run *ffmpeg*, `retriable-download` handled it in a single line of code

**Comment by codealchemist** [comment]
Hey @duykhang53 , I created this PR #34 to provide the metadata functionality.
I started playing with it and ended up with the PR.
Since you started it maybe you want to contribute testing it and commenting on it?

Cheers!

**Comment by codealchemist** [comment]
Done in #34

**Pull request closed without merging**